### ISSUE TITLE
Verify transitive dependencies

### ISSUE THREAD

**Issue by mweststrate** [mweststrate/relative-deps#6]
Currently, relative packages are not installed through `yarn` / `npm`, but simply extracted. While in general that works great, it might cause issues if there are differences between the transitive dependencies of the local package, and the published one. It would be great to detect and fix that.

**Comment by rifler** [comment]
`npm` supports installing tarball, so we can run install here - https://github.com/mweststrate/relative-deps/blob/master/index.js#L150

**Comment by mweststrate** [comment]
Yeah, biggest problem is that it makes the process very slow. So for
development purposes it is not really suitable imho. But probably we could
do something smart, like install from tarball only if package.json changed
or something.

Op ma 15 jul. 2019 21:56 schreef Dmitrii Kanatnikov <
notifications@github.com>:

**Comment by rifler** [comment]
> So for development purposes it is not really suitable imho

Most often yes, but what if package has deps that cannot be deduped?
```
node_modules
  lodash@3
  relative-dep
    node_modules
      lodash@2
```

I don't know how often does this happen. I suppose that for UI lib it is not good to have deps (better to have peerDeps and devDeps for tests), but for server lib it is quite normal situation. Maybe I am wrong

**Comment by mweststrate** [comment]
I think this can be quite simply supported: on first install / changed
package.json do a normal npm install of the tarball. On incremental
installs, remove everything but the node_modules and update from the
tarball.
